### PR TITLE
feat: New process image post functionality

### LINKED ISSUE #4: Add post image method
### Feature Description / Problem Solved by Feature

Add post image method for nexus

---

### Deliverables
- [x] add tests!
- [x] add process.functionalities.image.create  e.g.`API.post("images","processes/:id/functionalities/:id/images)`
- [x] add correct url parameters and body
- [x] update README

### PR DESCRIPTION
**Pull request by FelixMZ2018**
POSTs to `{processId}/functionality/{functionalityId}`, requires a body matching the following schema
```
{
imageData: base64encodedImage
fileType: fileType
}
```

### CODE REVIEW

File: src/process/functionality/image/index.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const create = require('./create')
+
+/**
+ * Uploads a process image, the image needs to be encoded in base64
+ * @module image
+ */
```
Comment (FelixMZ2018) [inline src/process/functionality/image/index.js]:
```suggestion
```

File: .eslintrc.js [hunk 2]
```diff
@@ -1,34 +1,44 @@
 module.exports = {
-  "extends": [
-    "airbnb-base",
-    "plugin:jest-formatting/recommended",
-    "plugin:jest/style",
-    "plugin:jest/all",
+  extends: [
```
Comment (DanielSeehausen) [inline .eslintrc.js:2]: 🙏

File: src/process/functionality/image/create.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const Amplify = require('aws-amplify')
+const serializeAxiosResponse = require('../../../utils/serializeAxiosResponse')
+const { DEFAULT_REQ_OPTS } = require('../../../utils/defaultReqOpts')
+/**
+ * Uploads a process image, the image needs to be encoded in base64
+ * requires the image to be encoded in base64
```
Comment (DanielSeehausen) [inline src/process/functionality/image/create.js]: perhaps duplicate description re: base 64

File: src/process/functionality/image/create.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+ * and the file type as string
+ * @param {string} ProcessId @param {string} FunctionalityID @param {object} Body
```
Comment (DanielSeehausen) [inline src/process/functionality/image/create.js]:
```suggestion
 * @param {string} ProcessId @param {string} FunctionalityId @param {object} Body
```

(or big D on processId)

**Review by DanielSeehausen — COMMENTED** [review]
Could we add this to the project if we are to start using jsdoc?
Once installed, the plugin might need to be added to the eslint config file

https://eslint.org/docs/rules/valid-jsdoc

**Review by DanielSeehausen — CHANGES_REQUESTED** [review]
looks great to me! 

only blocking changes are to 
- add jsdoc linting
- update test file path to match src file path

**Comment by FelixMZ2018** [comment]
> Could we add this to the project if we are to start using jsdoc?
> Once installed, the plugin might need to be added to the eslint config file
> 
> https://eslint.org/docs/rules/valid-jsdoc

I added https://github.com/gajus/eslint-plugin-jsdoc#eslint-plugin-jsdoc-installation since the eslint version is end of life and its the recommended replacement



> looks great to me!
> 
> only blocking changes are to
> 
> * add jsdoc linting
> * update test file path to match src file path

done and done

**Review by DanielSeehausen — APPROVED** [review]
gopher it!

**Pull request merged**